AnnoDesigner/AnnoCanvas.xaml.cs

```diff
@@ -1165,64 +1119,79 @@ protected override void OnMouseMove(MouseEventArgs e)
                                 }
-                                // adjust the drag start to compensate the amount we already moved
-                                _mouseDragStart.X += _coordinateHelper.GridToScreen(dx, GridSize);
-                                _mouseDragStart.Y += _coordinateHelper.GridToScreen(dy, GridSize);
-                            }
 
-                            StatisticsUpdated?.Invoke(this, EventArgs.Empty);
-                            break;
+                                var collisionsExist = false;
+                                // temporarily move each object and check if collisions with unselected objects exist
+                                foreach (var curLayoutObject in _selectedObjects)
+                                {
+                                    var originalPosition = curLayoutObject.Position;
+                                    // move object                                
+                                    curLayoutObject.Position = new Point(curLayoutObject.Position.X + dx, curLayoutObject.Position.Y + dy);
+                                    // check for collisions                                
+                                    var collides = _unselectedObjects.Find(_ => ObjectIntersectionExists(curLayoutObject, _)) != null;
+                                    curLayoutObject.Position = originalPosition;
+                                    if (collides)
+                                    {
+                                        collisionsExist = true;
+                                        break;
+                                    }
+                                }
+
+                                // if no collisions were found, permanently move all selected objects
+                                if (!collisionsExist)
+                                {
+                                    foreach (var curLayoutObject in _selectedObjects)
+                                    {
+                                        curLayoutObject.Position = new Point(curLayoutObject.Position.X + dx, curLayoutObject.Position.Y + dy);
+                                    }
+                                    // adjust the drag start to compensate the amount we already moved
+                                    _mouseDragStart.X += _coordinateHelper.GridToScreen(dx, GridSize);
+                                    _mouseDragStart.Y += _coordinateHelper.GridToScreen(dy, GridSize);
+
+                                    //position change -> update
+                                    StatisticsUpdated?.Invoke(this, new UpdateStatisticsEventArgs(UpdateMode.NoBuildingList));
+                                }
+
+                                //only update when positions were changed
+                                //StatisticsUpdated?.Invoke(this, EventArgs.Empty);
```

Comment:
Same as above: 

What is the purpose of this commented out code - if we don't need it then remove it, or add some additional explanation as to why its there.

Reply:
I removed it.